python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -29,6 +29,18 @@
 
 logger = Ganga.Utility.logging.getLogger(modulename=1)
 
+_imported_GangaList = None
+
+do_not_copy = ['_proxyObject', '_proxyClass', '_data', '_index_cache', '_parent', '_registry', '_impl']
+
+def _getGangaList():
+    global _imported_GangaList
+    if _imported_GangaList is None:
+        from Ganga.GPIDev.Lib.GangaList.GangaList import GangaList
+        _imported_GangaList = GangaList
+    return _imported_GangaList
+
+
```

Comment:
Really, all `GangaList` should be used for is to provide a proxy wrapper for a standard Python list. It's there so that we can store a list of raw objects behind the scenes and then when an object is requested, it will be proxy-wrapped automatically. Likewise, adding a proxied object should strip the proxy on its way in so the raw object is stored. This is needed because otherwise there's no way to provide a proxy interface which takes or returns a list. In principle `GangaObjects` should be storing internal things as python lists and not `GangaList`s.

Now, over the years this has been blurred and `GangaList` has been added all over the code where it is not needed and I'm trying to avoid making this worse.